### ISSUE TITLE
Consider swapping input params for RelayRecordProxy#setValue

### ISSUE THREAD

**Issue by mattkrick** [facebook/relay#1944]
In just about every piece of JS code, setters usually take the form of `key, value`. A few example are Map#set, everything in facebook's immutable-js, and even how we read objects `{key: 'value'}`. `RelayRecordProxy#setValue` asks for `value, key` & I often goof this up when i write mutations.

Before this stuff gets documented, I propose swapping the inputs for  `setValue, setLinkedRecord, setLinkedRecords`. It's an easy codemod & it'll make the API much easier to love :smile:

**Comment by josephsavona** [comment]
We debated this a lot internally during development of the API. The main reason we went with the current approach was that the `set*` functions accept a third parameter that is the arguments:

```
record.setValue(
  'https://...', 
  'uri', 
  {size: 128},
)
```

Reversing the key/value order means either: `setX(key, value, args)` (weird) or overloading the function arguments such that you do either `setX(key, value)` or `setX(key, args, value)`, where the meaning of the arguments depends on how many are passed (possible to verify with Flow, but hard to dynamically verify to give good errors for people not using Flow). That said - suggestions welcome for an API that addresses these concerns!

**Comment by mattkrick** [comment]
yeah, I definitely see where you're coming from  because params 2 & 3 really go together, just like they do with the getters. Personally, I'd say the overloaded args is intuitive enough because a signature like `foo(val, options, cb)` is pretty mainstream, but that's just my take.
